src/main/java/org/primefaces/component/api/UICalendar.java

```diff
@@ -153,19 +155,49 @@ public boolean hasTime() {
         return (pattern != null && (pattern.contains("HH") || pattern.contains("mm") || pattern.contains("ss")));
     }
 
+    /**
+     * date-only pattern
+     * @return
+     */
     public String calculatePattern() {
         String pattern = getPattern();
         Locale locale = calculateLocale(getFacesContext());
 
-        return pattern == null ? ((SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, locale)).toPattern() : pattern;
+        //return pattern == null ? ((SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, locale)).toPattern() : pattern;
+        if (pattern == null) {
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT);
+            return DateTimeFormatterBuilder.getLocalizedDateTimePattern(FormatStyle.SHORT, null, dateTimeFormatter.getChronology(), locale);
+        }
+        else return pattern;
+    }
+
+    public String calculateDateTimePattern() {
+        String pattern = getPattern();
+        Locale locale = calculateLocale(getFacesContext());
+
+        //return pattern == null ? ((SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, locale)).toPattern() : pattern;
+        if (pattern == null) {
+            DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT);
+            //TODO: timeStyle needs to be changed?
+            return DateTimeFormatterBuilder.getLocalizedDateTimePattern(FormatStyle.SHORT, FormatStyle.SHORT, dateTimeFormatter.getChronology(), locale);
+        }
+        else return pattern;
     }
 
     public String calculateTimeOnlyPattern() {
         if (timeOnlyPattern == null) {
-            String localePattern = ((SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, calculateLocale(getFacesContext()))).toPattern();
+            /*
+            Locale locale = calculateLocale(getFacesContext());
+            //TODO: SimpleDateFormat --> DateTimeFormatter
+            String localePattern = ((SimpleDateFormat) DateFormat.getDateInstance(DateFormat.SHORT, locale)).toPattern();
             String userTimePattern = getPattern();
 
             timeOnlyPattern = localePattern + " " + userTimePattern;
+             */
+
+            //TODO: Default-Pattern when no Pattern is specified via Pattern-Attribute?
```

Comment:
shouldn't we just add a default pattern in the stateHelper call?

Reply:
Compared to java.util.Date - Code and think the easy solution is equivalent to the old behavior.